diff --git a/core/modules/views/src/Hook/ThemeSuggestionsHook.php b/core/modules/views/src/Hook/ThemeSuggestionsHook.php
new file mode 100644
index 0000000000000000000000000000000000000000..5da73fd13b154694dd946ae51fd52e2d0097c786
--- /dev/null
+++ b/core/modules/views/src/Hook/ThemeSuggestionsHook.php
@@ -0,0 +1,27 @@
+<?php
+
+namespace Drupal\views\Hook;
+
+use Drupal\Core\Hook\Attribute\Hook;
+
+/**
+ * Theme suggestions for views.
+ */
+class ThemeSuggestionsHook {
+
+  /**
+   * Implements hook_theme_suggestions_HOOK().
+   */
+  #[Hook('theme_suggestions_views_view')]
+  public function themeSuggestionsViewsView(array $variables): array {
+    $suggestions = [];
+    $view = $variables['view'];
+
+    $suggestions[] = 'views_view__' . $view->id();
+    $suggestions[] = 'views_view__' . $view->current_display;
+    $suggestions[] = 'views_view__' . $view->id() . '__' . $view->current_display;
+
+    return $suggestions;
+  }
+
+}
