diff --git a/core/modules/rest/src/Plugin/views/style/Serializer.php b/core/modules/rest/src/Plugin/views/style/Serializer.php
index b2dc484..c7c0391 100644
--- a/core/modules/rest/src/Plugin/views/style/Serializer.php
+++ b/core/modules/rest/src/Plugin/views/style/Serializer.php
@@ -86,6 +86,7 @@ public function __construct(array $configuration, $plugin_id, $plugin_definition
   protected function defineOptions() {
     $options = parent::defineOptions();
     $options['formats'] = ['default' => []];
+    $options['pager'] = ['default' => FALSE];
 
     return $options;
   }
@@ -103,6 +104,12 @@ public function buildOptionsForm(&$form, FormStateInterface $form_state) {
       '#options' => $this->getFormatOptions(),
       '#default_value' => $this->options['formats'],
     ];
+    $form['pager'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Add pager information'),
+      '#description' => $this->t('Pager information will be added to rest export. <b>Output structure will be changed.</b>'),
+      '#default_value' => $this->options['pager'],
+    ];
   }
 
   /**
@@ -139,6 +146,30 @@ public function render() {
     else {
       $content_type = !empty($this->options['formats']) ? reset($this->options['formats']) : 'json';
     }
+
+    $pager = $this->view->pager;
+    if ($this->options['pager'] && $pager && $this->displayHandler && get_class($this->displayHandler) === 'Drupal\rest\Plugin\views\display\RestExport') {
+      $class = get_class($pager);
+      $current_page = $pager->getCurrentPage();
+      $items_per_page = $pager->getItemsPerPage();
+      $total_items = $pager->getTotalItems();
+
+      $total_pages = 0;
+      if (!in_array($class, ['Drupal\views\Plugin\views\pager\None', 'Drupal\views\Plugin\views\pager\Some'])) {
+        $total_pages = $pager->getPagerTotal();
+      }
+      $result = [
+        'rows' => $rows,
+        'pager' => [
+          'current_page' => $current_page,
+          'total_items' => $total_items,
+          'total_pages' => $total_pages,
+          'items_per_page' => $items_per_page,
+        ],
+      ];
+      return $this->serializer->serialize($result, $content_type, ['views_style_plugin' => $this]);
+    }
+
     return $this->serializer->serialize($rows, $content_type, ['views_style_plugin' => $this]);
   }
 
